### PR TITLE
be able to transfer karma/butts between names

### PR DESCRIPTION
**Pull request by himmelmaus**
as referenced in issue #42

### CODE REVIEW

File: ircbot/plugins/transfer.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    if sender.lower() == bot.config['System']['owner']:
+        redis = StrictRedis.from_url(bot.config['System']['redis_url'])
+
+        if len(args) == 2:
+            oldname = args[0].lower()
+            newname = args[1].lower()
+        else:
+            string = ""
+            strings = []
+            for x in args:
```
Comment (neanias) [inline ircbot/plugins/transfer.py]: `x` is not a clear variable name.

File: ircbot/plugins/transfer.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                string = string + " " + x
+            for y in re.findall("(?![\'\"])[a-z,\s]+(?![\"\`])", string):
```
Comment (neanias) [inline ircbot/plugins/transfer.py]: Nor is `y`.

File: ircbot/plugins/transfer.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+@bot.command('transfer')
+def transfer(bot, channel, sender, args):
+    """"command to transfer karma & butts from one term to another"""
+    if sender.lower() == bot.config['System']['owner']:
+        redis = StrictRedis.from_url(bot.config['System']['redis_url'])
+
+        if len(args) == 2:
+            oldname = args[0].lower()
+            newname = args[1].lower()
```
Comment (neanias) [inline ircbot/plugins/transfer.py]: These should both be underscored: `old_name`, `new_name`.

File: ircbot/plugins/transfer.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        else:
+            string = ""
+            strings = []
+            for x in args:
+                string = string + " " + x
```
Comment (neanias) [inline ircbot/plugins/transfer.py]: It would probably be cleaner to use a format string here.

**Review by neanias — CHANGES_REQUESTED** [review]

File: ircbot/plugins/transfer.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        except TypeError:
+            old_butts = 0
+        try:
+            new_butts = current = int(redis.hget(bot.config['System']['redis_prefix'] + "buttmaster", new_name))
+        except TypeError:
+            new_butts = 0
+        redis.hset(bot.config['System']['redis_prefix'] + "buttmaster", new_name, new_butts + old_butts)
+        redis.hdel(bot.config['System']['redis_prefix'] + "buttmaster", old_name)
+        
+        bot.message(channel, "{} now has {} karma and {} butts, {} now has no karma and no butts".format(new_name,old_karma+new_karma,old_butts+new_butts,old_name))
```
Comment (neanias) [inline ircbot/plugins/transfer.py]: This line is **way** too long

File: ircbot/plugins/transfer.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        redis = StrictRedis.from_url(bot.config['System']['redis_url'])
+
+        if len(args) == 2:
+            old_name = args[0].lower()
+            new_name = args[1].lower()
+        else:
+            string = ""
+            strings = []
+            for word in args:
+                string = "{} {}".format(string, word)
```
Comment (neanias) [inline ircbot/plugins/transfer.py]: What are you doing here, exactly? Are you just joining all the `args` together?

**Review by neanias — CHANGES_REQUESTED** [review]
More comments. You're getting there, keep going!

File: ircbot/plugins/transfer.py (hunk 5 above)
Comment (himmelmaus) [reply ircbot/plugins/transfer.py]: line 49 I presume?

Comment (neanias) [reply ircbot/plugins/transfer.py]: Yes

File: ircbot/plugins/transfer.py (hunk 6 above)
Comment (himmelmaus) [reply ircbot/plugins/transfer.py]: yes, so the regex can deal with it

File: ircbot/plugins/transfer.py (hunk 5 above)
Comment (himmelmaus) [reply ircbot/plugins/transfer.py]: go big or go home

File: ircbot/plugins/transfer.py (hunk 6 above)
Comment (neanias) [reply ircbot/plugins/transfer.py]: Python has a `join` function for lists to join them together into strings. I'd recommand changing this to `strings = args.join(" ")`.

File: ircbot/plugins/transfer.py (hunk 5 above)
Comment (himmelmaus) [reply ircbot/plugins/transfer.py]: but how am I meant I fix that? just splitting the arguments across several lines?

Comment (neanias) [reply ircbot/plugins/transfer.py]: Tell that to the linter

Comment (neanias) [reply ircbot/plugins/transfer.py]: Yes

File: ircbot/plugins/transfer.py (hunk 6 above)
Comment (himmelmaus) [reply ircbot/plugins/transfer.py]: oh yeah it does doesn't it, ok I'll add that

**Comment by himmelmaus** [comment]
@neanias are you happy with that in its current state?

File: ircbot/plugins/transfer.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        for term in re.findall("(?![\'\"])[a-z,\s]+(?=[\"\'])", string):
+            strings.append(term.strip())
+        try:
+            first_name = strings[0].lower()
+            second_name = strings[1].lower()
+        except IndexError:
+            bot.message(channel, "Oops! Looks like you didn't give enough arguments!")
+            return
+
+    if sender.lower() == bot.config['System']['owner']:
```
Comment (hreeder) [inline ircbot/plugins/transfer.py]: IMO, this should come earlier - why perform the searching if the user isn't authorized to run the command?

File: ircbot/plugins/transfer.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    else:
+        string = args.join(" ")
+        strings = []
+        for term in re.findall("(?![\'\"])[a-z,\s]+(?=[\"\'])", string):
+            strings.append(term.strip())
+        try:
+            first_name = strings[0].lower()
+            second_name = strings[1].lower()
+        except IndexError:
+            bot.message(channel, "Oops! Looks like you didn't give enough arguments!")
```
Comment (hreeder) [inline ircbot/plugins/transfer.py]: Maybe echo the usage here to prompt the user as to how it should look?

File: ircbot/plugins/transfer.py [hunk 9]
```diff
@@ -0,0 +1,55 @@
+import re
+
+from ircbot import bot
+from redis import StrictRedis
+
+
+@bot.command('transfer')
+def transfer(bot, channel, sender, args):
+    """"command to transfer karma & butts from one term to another"""
```
Comment (hreeder) [inline ircbot/plugins/transfer.py]: Docstring should contain usage for something like this

**Review by hreeder — CHANGES_REQUESTED** [review]

File: ircbot/plugins/transfer.py [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+    if len(args) == 2:
+            first_name = args[0].lower()
+            second_name = args[1].lower()
+    else:
+        string = args.join(" ")
```
Comment (hreeder) [inline ircbot/plugins/transfer.py]: Iirc this should be `" ".join(args)`

File: ircbot/plugins/transfer.py (hunk 7 above)
Comment (himmelmaus) [reply ircbot/plugins/transfer.py]: oh, I had that originally but changed it for when I added the permission to be able to transfer your own karma

File: ircbot/plugins/transfer.py (hunk 8 above)
Comment (himmelmaus) [reply ircbot/plugins/transfer.py]: will do

File: ircbot/plugins/transfer.py (hunk 10 above)
Comment (neanias) [reply ircbot/plugins/transfer.py]: (sorry, my bad, I forgot how to Python)

**Review by hreeder — APPROVED** [review]

**Pull request merged**